pipeline-runner/R/gem2s-4-score_doublets.R

```diff
@@ -1,65 +1,61 @@
-#  - Compute doublet scores per sample
-#  - Save an txt file with doublet scores per sample [doublet-scores-sampleName]
-
-score_doublets <- function(input, pipeline_config) {
-  print(list.files(paste("/output", sep = "/"), all.files = TRUE, full.names = TRUE, recursive = TRUE))
-  message("reloading old matrices...")
-
-  scdata_list <- readRDS("/output/pre-doublet-scdata_list.rds")
-
-  message("Loading configuration...")
-  config <- RJSONIO::fromJSON("/input/meta.json")
-
-  print_config(4, "Score Doublets", input, pipeline_config, config)
-
-  # Check which samples have been selected. Otherwiser we are going to use all of them.
-  if (length(config$samples) > 0) {
-    samples <- config$samples
-  } else {
-    samples <- names(scdata_list)
+#' Compute doublet score for count matrices
+#'
+#' @inheritParams download_cellranger
+#' @param prev_out 'output' slot from call to \code{run_emptydrops}
+#'
+#' @return \code{prev_out} with added slot 'doublet_scores' containing result of call to
+#'   \code{\link{compute_doublet_scores}} for each sample.
+#'
+#' @export
+#'
+score_doublets <- function(input, pipeline_config, prev_out) {
+  message("Calculating probability of droplets being doublets...")
+
+  edrops_list <- prev_out$edrops
+  counts_list <- prev_out$counts_list
+  samples <- names(counts_list)
+
+  scores <- list()
+  for (sample in samples) {
+    message("Sample --> ", sample, "...")
+    counts <- counts_list[[sample]]
+    edrops <- edrops_list[[sample]]
+
+    # scDblFinder expects empty droplets to be removed
+    if (!is.null(edrops)) {
+      keep <- which(edrops$FDR <= 0.001)
```

Comment:
better make an R object or whatever it is called and add all those settings that we are using for gem2s, then get this 0.001 number from the object instead of hardcoding it